pykusto/expressions.py

```diff
@@ -44,7 +43,8 @@ def __new__(cls, *args, **kwargs):
     def __init__(self, kql: Union[KQL, 'BaseExpression']) -> None:
         if isinstance(kql, BaseExpression):
             self.kql = kql.kql
-        elif not isinstance(kql, str):
```

Comment:
Because for a `BaseExpression` instance the last line of the method would run, while it was not supposed to. Just a plain old bug (one of the new tests I added fails if you revert this change).